subprojects/plugins/src/main/java/org/gradle/jvm/application/tasks/CreateStartScripts.java

```diff
@@ -253,7 +250,8 @@ public Property<String> getMainClass() {
     /**
      * The main class name used to start the Java application.
      */
-    @ReplacedBy("mainClass")
+    @Input
+    @Optional
     @Nullable
     @Deprecated
```

Comment:
How come these methods are also not being removed?

Reply:
These are still used by heavily used plugins under smoke test, so can't be removed now.  We'll update the deprecation warning to reference Gradle 9 when we remove all the Gradle 8 related warnings.  Hopefully by Gradle 9 new versions of the plugins will be available which have completely migrated away from these and they can be removed then, but for now, they must stay.